drivers/adodb-mysqli.inc.php

```diff
@@ -231,9 +231,18 @@ function _connect($argHostname = null,
 
 		// SSL Connections for MySQLI
 		if ($this->ssl_key || $this->ssl_cert || $this->ssl_ca || $this->ssl_capath || $this->ssl_cipher) {
+
 			mysqli_ssl_set($this->_connectionID, $this->ssl_key, $this->ssl_cert, $this->ssl_ca, $this->ssl_capath, $this->ssl_cipher);
-			$this->socket = MYSQLI_CLIENT_SSL;
-			$this->clientFlags = MYSQLI_CLIENT_SSL_DONT_VERIFY_SERVER_CERT;
+
+			// Check for set SSL client flag, NOTE: bitwise operations.
+			$ssl_client = ($this->clientFlags & MYSQLI_CLIENT_SSL) == MYSQLI_CLIENT_SSL;
+			$ssl_verify_cert = ($this->clientFlags & MYSQLI_CLIENT_SSL_VERIFY_SERVER_CERT) == MYSQLI_CLIENT_SSL_VERIFY_SERVER_CERT;
+			$ssl_dont_verify_cert = ($this->clientFlags & MYSQLI_CLIENT_SSL_DONT_VERIFY_SERVER_CERT) == MYSQLI_CLIENT_SSL_DONT_VERIFY_SERVER_CERT;
```

Comment:
The `== MYSQLI_CLIENT_xxxx` bit is not necessary, the bitwise AND will return either 0 or a non-zero value, which PHP will evaluate as TRUE.